### DISCUSSION TITLE
Inside custom API I want to update user

### DISCUSSION THREAD

**Discussion by harshalone** [directus/directus#9769] [Archive]
hi I have created a custom API

```
basically : DOMAIN/verify_email
payload:
{
 "user_id" : "234234234324",
"code" : "123456"
}
```

What I want to do is to check if this code is correct from users table and then update my users status to active.

How do I do that?

Like this:

`users.items.update({ "ACTIVE" }, { "status" }, directus_users);`

Help will be appriciated

**Comment by rijkvanzanten** [comment]
When you say you created a custom API, do you mean a custom endpoint? 

If you're saying that you built a totally custom API, I'd recommend using like the SDK or axios to talk to Directus:

```js
const userInfo = await users.readOne(payload.user_id, { fields: ['code'] });

if (userInfo.data.code === payload.code) {
  await users.updateOne(payload.user_id, { status: 'active' });
}
```

if you're talking about a custom endpoint, you can use the UsersService as follows:

```js
export default (router, { services }) => {
  const { UsersService } = services;

  router.get('/verify_email', async (req, res) => {
    const usersService = new UsersService({ schema: req.schema });
    const userInfo = await usersService.readOne(req.body.user_id, { fields: ['code'] });

    if (userInfo.data.code === req.body.code) {
      await usersService.updateOne(payload.user_id, { status: 'active' });
    }
  });
});
